docs/Model Pipeline.md

```diff
@@ -0,0 +1,351 @@
+---
+author: Asim Hussain (@jawache)
+abstract: How models are chained together into a pipeline in order to compute the impacts of
+---
+# Model Pipeline
+
+## Introduction
+
+As [Peter H. Salus](https://en.wikipedia.org/wiki/Peter_H._Salus "Peter H. Salus") said in his book [A Quarter-Century of Unix](https://en.wikipedia.org/wiki/Unix_philosophy#cite_note-taoup-ch1s6-1) the Unix philosophy is:
+- Write programs that do one thing and do it well.
+- Write programs to work together.
+- Write programs to handle text streams, because that is a universal interface.
+
+Our approach to models in the [Impact Engine Framework](Impact%20Engine%20Framework.md) is the same. 
+- Each model will do one thing and do it well. 
+- Models work with each other. 
+- The [Impl (Impact YAML)](Impl%20(Impact%20YAML).md) text format is the communication medium between models.
+
+Calculating a component's impacts often requires using multiple models in sequence. Each model takes as input the outputs of the previous model in the chain, all working together to calculate impacts from observations.
+
+```mermaid 
+flowchart LR
+Observations --> Model1 --> Model2 --> Model3 --> Impacts
+```
+
+At the start of the chain, we input source observations. Each model in the chain takes as input the observations and modifies them somehow before passing them along to the next model in the chain.
+
+The nature of the modification is flexible and defined by the model. Some model plugins will calculate an impact metric (for example, energy from utilization) and add that to the observation before passing that to the next model. Some models will enrich the observation with meta-data required for other models, for example, adding grid carbon intensity values.
+
+## Initialization
+
+All the model plugins used in any component in the graph are configured at the top of the [Impl (Impact YAML)](Impl%20(Impact%20YAML).md) in the initialize section. There are multiple places to add configuration for model plugins; any configuration added here is global and applied to every use of the model throughout the graph.
+
+```yaml
+.
+.
+initialize:
+  plugins:
+    - name: <model-name-1>
+      path: <path-to-ts-module-to-load>
+      config:
+	       <key>: <value>
+    - name: <model-name-2>
+      path: <path-to-ts-module-to-load>
+      config:
+	       <key>: <value>	       
+.
+.
+graph: ~
+```
+
+## Usage
+
+In component nodes, we then configure the models we want to use in a pipeline like so:
+
+```yaml
+backend:
+  pipeline: "model-1 | model-2 | model-3"
+  config: 

```

Comment:
maybe here we can define the functional unit and the operation required to represent the SCI in those units. For example, several of the current case studies calculate the SCI per s/hour/month and then retroactively apply some scaling to transform it into a SCI per API request or per website visit. This could be captured as follows:

```
config:
    funit:
        - name: request
          scale-factor: 32000 # divide SCI (per hour) by this value to get SCI per api call
```

Reply:
Yeah I pulled back from describing the sci approach using a pipeline. It gets complex! But let's add this context in another doc just about how to implement the SCI